[BUG] Build error on M1 when running pnpm prep

**Describe the bug**
> `pnpm prep` is producing a build error on M1 MBP

It's entirely possible this is an environmental issue and not related to M1 directly, I'm not a Rust developer, this was a clean install of Rust.

**To Reproduce**
1. Have an M1 MBP
2. Follow contributing guide
3. At the `pnpm prep` step, see error

**Output**
```yaml
error: linking with `cc` failed: exit status: 1
  |
  = note: "cc" "-arch" "arm64" "/repo/spacedrive/target/debug/deps/sdcore-057366575273bd28.11l54r2yneiizk98.rcgu.o" ... 
  = note: ld: in /Users/JohnDoe/Documents/GitHub/spacedrive/target/debug/deps/libring-51854f980945fc0e.rlib(chacha-armv8-ios64.o), building for macOS, but linking in object file built for iOS, for architecture arm64
          clang: error: linker command failed with exit code 1 (use -v to see invocation)      

warning: `sdcore` (lib test) generated 4 warnings (4 duplicates)
error: could not compile `sdcore` due to previous error; 4 warnings emitted
warning: build failed, waiting for other jobs to finish...
warning: `sdcore` (lib) generated 4 warnings
error: build failed
/Users/JohnDoe/Documents/GitHub/spacedrive/core:
 ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @sd/core@0.0.0 codegen: `cargo test && ts-node ./scripts/bindingsIndex.ts`
Exit status 101
```

--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -1,2 +1,8 @@
 [alias]
 prisma = "run -p prisma-cli --"
+
+[target.x86_64-apple-darwin]
+rustflags = [
+  "-C", "link-arg=-undefined",
+  "-C", "link-arg=dynamic_lookup",
+]

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -35,6 +35,8 @@ Scan through our [existing issues](https://github.com/github/docs/issues) to fin
 
 This project uses [Cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html) and [pnpm](https://pnpm.io/installation). Ensure you have them installed before continuing.
 
+> Note: MacOS M1 users should choose the customize option in the rustup init script and enter `x86_64-apple-darwin` as the default host triple instead of the default `aarch64-apple-darwin`
+
 - `$ git clone https://github.com/spacedriveapp/spacedrive`
 - For Linux or MacOS users run: `chmod +x ./.github/scripts/setup-system.sh && ./.github/scripts/setup-system.sh`
   - This will install FFMPEG and any other required dependencies for Spacedrive to build.

